### PR TITLE
fix: login commnad with --api-token #96

### PR DESCRIPTION
**Pull request by vashian**

### CODE REVIEW

File: src/utils/liara-api-request.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  }
+
+  async login(options: Record<string, any>) {
+    const { fullname, avatar, api_token } = (await retry(async () => {
+      try {
+        return await this.httpClient
+          .post('v1/login', {
+            json: options.body,
+            headers: {
+              Authorization: options.api_token
```
Comment (mhemrg) [inline src/utils/liara-api-request.ts:56]: We don't need api token for the `/login` endpoint at all.

File: src/commands/account/add.ts [hunk 2]
```diff
@@ -64,41 +65,36 @@ export default class AccountAdd extends Command {
...
+          api_token: flags['api-token'],
+        })
+      : await liaraAPI.login({
+          body,
+          api_token: flags['api-token'],
+          region,
+        });
+
+    if (flags.email !== account.email) {
+      this.error(`Email: ${flags.email} is not related to your api-token`);
```
Comment (mhemrg) [inline src/commands/account/add.ts:92]: How may this happen?

**Comment by mhemrg** [comment]
A lot of changes have been applied in this PR. Can you also take a look at, please? @guoard

And I'm cool with the idea of separating API calls to a new `Liara` class. We can extract this class to it's own package in the future and release it as `@liara/client` so everyone can interact with our REST API easily.

**Comment by guoard** [comment]
@vashian Shall you please rebase `master` branch and resolve conflict first?

**Pull request closed without merging**